org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/MavenPluginActivator.java

```diff
@@ -234,7 +236,8 @@ public void start(final BundleContext context) throws Exception {
     this.configurationManager = new ProjectConfigurationManager(maven, managerImpl, modelManager, mavenMarkerManager,
         mavenConfiguration);
     this.projectManager.addMavenProjectChangedListener(this.configurationManager);
-    workspace.addResourceChangeListener(configurationManager, IResourceChangeEvent.PRE_DELETE);
+    context.registerService(IResourceChangeListener.class, configurationManager,
```

Comment:
What about the removeResourceListener then? Shouldn't we unregister the service instead?

Reply:
All services are automatically disposed when the bundle is stopped.

Reply:
So the removeResourceListener can be removed?

Reply:
Of course yes, I have updated the PR